lib/index.js

```diff
@@ -108,15 +108,12 @@ api.set = async ({
     // FIXME: may need to allow up to N tokens, one per `clientId` if
     // `clientId` is set
 
-    // if there is an existing, unexpired nonce, reuse it, do not generate
-    // a new one to avoid DoS attack of constantly running bcrypt algorithm
+    // if there is an existing nonce, delete it, and generate a new one
     try {
-      const existingToken = await api.get({actor: null, account, email, type});
-      // keep existing token if it hasn't expired; note that there is no
-      // `challenge` to return because it is not known in the database
-      if(existingToken && new Date() < new Date(existingToken.expires)) {
-        return {type};
-      }
+      // check to see if nonce exists
+      await api.get({actor: null, account, email, type});
+      // the `get` API call did not throw, a nonce exists, delete it
+      await api.remove({actor, account, type});
```

Comment:
This may need to change based on the above comments anyway -- but what happens if two or more requests that are made concurrently hit this code path?